datasets/emotion/emotion.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+import nlp
+
+
+_CITATION = """\
+@inproceedings{saravia-etal-2018-carer,
+    title = "{CARER}: Contextualized Affect Representations for Emotion Recognition",
+    author = "Saravia, Elvis  and
+      Liu, Hsien-Chi Toby  and
+      Huang, Yen-Hao  and
+      Wu, Junlin  and
+      Chen, Yi-Shin",
+    booktitle = "Proceedings of the 2018 Conference on Empirical Methods in Natural Language Processing",
+    month = oct # "-" # nov,
+    year = "2018",
+    address = "Brussels, Belgium",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/D18-1404",
+    doi = "10.18653/v1/D18-1404",
+    pages = "3687--3697",
+    abstract = "Emotions are expressed in nuanced ways, which varies by collective or individual experiences, knowledge, and beliefs. Therefore, to understand emotion, as conveyed through text, a robust mechanism capable of capturing and modeling different linguistic nuances and phenomena is needed. We propose a semi-supervised, graph-based algorithm to produce rich structural descriptors which serve as the building blocks for constructing contextualized affect representations from text. The pattern-based representations are further enriched with word embeddings and evaluated through several emotion recognition tasks. Our experimental results demonstrate that the proposed method outperforms state-of-the-art techniques on emotion recognition tasks.",
+}
+"""
+
+_DESCRIPTION = """\
+Emotion is a dataset of English Twitter messages with eight basic emotions: anger, anticipation,
+disgust, fear, joy, sadness, surprise, and trust. For more detailed information please refer to the
+paper.
+"""
+_URL = "https://github.com/dair-ai/emotion_dataset"
+# use dl=1 to force browser to download data instead of displaying it
+_TRAIN_DOWNLOAD_URL = "https://www.dropbox.com/s/1pzkadrvffbqw6o/train.txt?dl=1"
+_VALIDATION_DOWNLOAD_URL = "https://www.dropbox.com/s/2mzialpsgf9k5l3/val.txt?dl=1"
+_TEST_DOWNLOAD_URL = "https://www.dropbox.com/s/ikkqxfdbdec3fuj/test.txt?dl=1"
+
+
+class EmotionConfig(nlp.BuilderConfig):
+    """BuilderConfig for Emotion dataset."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for Emotion dataset.
+
+    Args:
+      **kwargs: keyword arguments forwarded to super.
+    """
+        super(EmotionConfig, self).__init__(version=nlp.Version("0.1.0", "First Emotion release"), **kwargs)
+
+
+class Emotion(nlp.GeneratorBasedBuilder):
+
+    BUILDER_CONFIGS = [
+        EmotionConfig(
+            name="emotion",
+            description="Emotion classification dataset. Twitter messages are classifier as either anger, anticipation, disgust, fear, joy, sadness, surprise, or trust",
+        )
+    ]
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            description=_DESCRIPTION,
+            features=nlp.Features({"text": nlp.Value("string"), "label": nlp.Value("string")}),
+            homepage=_URL,
+            citation=_CITATION,
```

Comment:
`supervised_keys` can probably be set here:
```python
supervised_keys=("text", "label")
```

Reply:
thanks, added!